### PR TITLE
[doc] Add rdocs for RedisMemo::MemoizeQuery

### PR DESCRIPTION
**Pull request by katyho**
### Summary
Add rdocs for memoize_query

### Testing
After making all other methods as private, specs still pass

### CODE REVIEW

**Comment by katyho** [comment]
Need to fix build, also probably want to leave `RedisMemo::MemoizeQuery.invalidate` methods as public methods

File: lib/redis_memo/memoize_query.rb [hunk 1]
```diff
@@ -9,10 +9,35 @@ module RedisMemo::MemoizeQuery
...
+  #   - User.find_by_first_name(first_name)
+  # will first check the Redis cache for the data before hitting the SQL database;
+  # the cache results are invalidated automatically when user records are changed.
+  #
+  # Note that +memoize_table_column :first_name, last_name+ specifies that only AND queries
+  # that contain both columns will be memozed. The query +User.where(last_name: last_name)+
+  # will NOT be memoized with the given configuration.
+  #
+  # @param raw_columns [Array] A list of columns to memoize.
+  # @param ediable [Boolean] Specify if the column is editable. Only editable columns
```
Comment (donaldong) [inline lib/redis_memo/memoize_query.rb:37]: typo
```suggestion
  # @param editable [Boolean] Specify if the column is editable. Only editable columns
```
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia3.giphy.com%2Fmedia%2F12uXi1GXBibALC%2Fgiphy.gif&token=food" data-gitmeme-token="food"></a>

File: lib/redis_memo/memoize_query.rb [hunk 2]
```diff
@@ -9,10 +9,35 @@ module RedisMemo::MemoizeQuery
...
+  #   - User.find(user_id)
+  #   - User.where(id: user_id).first
+  #   - User.where(first_name: first_name).first
+  #   - User.where(first_name: first_name, last_name: last_name).first
+  #   - User.find_by_first_name(first_name)
+  # will first check the Redis cache for the data before hitting the SQL database;
+  # the cache results are invalidated automatically when user records are changed.
+  #
+  # Note that +memoize_table_column :first_name, last_name+ specifies that only AND queries
+  # that contain both columns will be memozed. The query +User.where(last_name: last_name)+
```
Comment (donaldong) [inline lib/redis_memo/memoize_query.rb:33]:
```suggestion
  # that contain both columns will be memoized. The query +User.where(last_name: last_name)+
```

File: lib/redis_memo/memoize_query.rb [hunk 3]
```diff
@@ -9,10 +9,35 @@ module RedisMemo::MemoizeQuery
   require_relative 'memoize_query/invalidation'
   require_relative 'memoize_query/model_callback'
 
-  # Only editable columns will be used to create memos that are invalidatable
-  # after each record save
+  # Core entry method for using RedisMemo to cache SQL select queries on the given
+  # column names.
```
Comment (donaldong) [inline lib/redis_memo/memoize_query.rb:13]: Let's link the CachdSelect doc here, and briefly describe how it's done.

E.g. "we intercept any activerecord select queries and extract the dependencies from SQL query parameters. learn more in +CachedSelect+"

**Review by donaldong — APPROVED** [review]
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia3.giphy.com%2Fmedia%2F3o7abB06u9bNzA8lu8%2Fgiphy.gif&token=lgtm" data-gitmeme-token="lgtm"></a>

**Pull request merged**